distribution/kernel/carbon-home/repository/resources/conf/default.json

```diff
@@ -140,6 +140,7 @@
   "encryption.algorithm": "AES",
   "encryption.alias": "symmetric.key.value",
   "encryption.internal_crypto_provider": "org.wso2.carbon.crypto.provider.KeyStoreBasedInternalCryptoProvider",
+  "encryption.enable_key_id": true,
```

Comment:
Why don't we need `old_key` in the default values, given that the key_id is enabled ?

My understanding is
If key ID is enabled, the decryption uses the key id of the secret, hence require the old key. Am I correct ?

Reply:
Or old secrets are not related to the key_id ?

Reply:
> Why don't we need `old_key` in the default values, given that the key_id is enabled ?
> 
> My understanding is If key ID is enabled, the decryption uses the key id of the secret, hence require the old key. Am I correct ?

During decryption, if a `key_id` is present in the encrypted value, the system first checks it against the `new_secret`. If it matches, the `new_secret` is used; otherwise, it falls back to the `old_secret`. The default value of `old_secret` is same as the value of `new_secret`, which is dynamically set at runtime.

> Or old secrets are not related to the key_id ?

Key_id is stored with the encrypted value during encryption, so it is not related to old secret.